core/src/Kokkos_Layout.hpp

```diff
@@ -222,6 +222,39 @@ template <typename Layout, class Enable = void>
 struct KOKKOS_DEPRECATED is_layouttiled : std::false_type {};
 #endif
 
+// The layout_iterate_type_selector is now deprecated.
+// Please use the Kokkos::Impl::layout_iterate_type_selector instead.
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+template <typename... Layout>
+struct KOKKOS_DEPRECATED layout_iterate_type_selector {
+  static const Kokkos::Iterate outer_iteration_pattern =
+      Kokkos::Iterate::Default;
+  static const Kokkos::Iterate inner_iteration_pattern =
+      Kokkos::Iterate::Default;
+};
+
+template <>
+struct KOKKOS_DEPRECATED layout_iterate_type_selector<Kokkos::LayoutRight> {
```

Comment:
IIRC we need to be careful with specializations of deprecated classes, some compilers warn about it.

Reply:
https://godbolt.org/z/T1f5KczGb

Reply:
We can just deprecate the `static const` members instead. That doesn't trigger these spurious warnings.

Reply:
I would do a deprecated alias https://godbolt.org/z/E7cnTMPzq